builtin/providers/cobbler/resource_cobbler_distro.go

```diff
@@ -0,0 +1,225 @@
+			},
+
+			"os_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"owners": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+
+			"redhat_management_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"redhat_management_server": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"template_files": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDistroCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Create a cobblerclient.Distro
+	distro := buildDistro(d, config)
+
+	// Attempte to create the Distro
+	log.Printf("[DEBUG] Cobbler Distro: Create Options: %#v", distro)
+	newDistro, err := config.cobblerClient.CreateDistro(distro)
+	if err != nil {
+		return fmt.Errorf("Cobbler Distro: Error Creating: %s", err)
+	}
+
+	d.SetId(newDistro.Name)
+
+	return resourceDistroRead(d, meta)
+}
+
+func resourceDistroRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Retrieve the distro from cobbler
+	distro, err := config.cobblerClient.GetDistro(d.Id())
+	if err != nil {
+		return fmt.Errorf("Cobbler Distro: Error Reading (%s): %s", d.Id(), err)
+	}
+
+	// Set all fields
+	d.Set("arch", distro.Arch)
+	d.Set("breed", distro.Breed)
+	d.Set("boot_files", distro.BootFiles)
+	d.Set("comment", distro.Comment)
+	d.Set("fetchable_files", distro.FetchableFiles)
+	d.Set("kernel", distro.Kernel)
+	d.Set("kernel_options", distro.KernelOptions)
+	d.Set("kernel_options_post", distro.KernelOptionsPost)
+	d.Set("initrd", distro.Initrd)
+	d.Set("mgmt_classes", distro.MGMTClasses)
+	d.Set("os_version", distro.OSVersion)
+	d.Set("owners", distro.Owners)
+	d.Set("redhat_management_key", distro.RedHatManagementKey)
+	d.Set("redhat_management_server", distro.RedHatManagementServer)
+	d.Set("template_files", distro.TemplateFiles)
+
+	return nil
+}
+
+func resourceDistroUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// create a cobblerclient.Distro
+	distro := buildDistro(d, config)
+
+	// Attempt to updateh the distro with new information
+	log.Printf("[DEBUG] Cobbler Distro: Updating Distro (%s) with options: %+v", d.Id(), distro)
+	err := config.cobblerClient.UpdateDistro(&distro)
+	if err != nil {
+		return fmt.Errorf("Cobbler Distro: Error Updating (%s): %s", d.Id(), err)
+	}
+
+	return resourceDistroRead(d, meta)
+}
+
+func resourceDistroDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Attempt to delete the distro
+	if err := config.cobblerClient.DeleteDistro(d.Id()); err != nil {
+		return fmt.Errorf("Cobbler Distro: Error Deleting (%s): %s", d.Id(), err)
+	}
+
+	d.SetId("")
```

Comment:
This is actually [done for you](https://github.com/hashicorp/terraform/blob/cd6d7328d4361712efb3c951690866d145f751e1/helper/schema/resource.go#L125-L126) if you return successfully from delete. :+1: